hasher-matcher-actioner/hmalib/common/config.py

```diff
@@ -322,10 +328,42 @@ def update_config(config: HMAConfig) -> None:
     # TODO - we should probably sanity check here to make sure all the fields
     #        are the expected types, because lolpython. Otherwise, it will
     #        fail to deserialize later
-    get_dynamodb().meta.client.put_item(
+    if insert_only:
+        try:
+            get_dynamodb().meta.client.put_item(
+                TableName=_TABLE_NAME,
+                Item=_config_to_dynamodb_item(config),
+                ConditionExpression=Attr(key).not_exists(),
+            )
+        except ClientError as e:
+            # Ignore the ConditionalCheckFailedException, bubble up
+            # other exceptions.
+            if e.response["Error"]["Code"] != "ConditionalCheckFailedException":
```

Comment:
This appears to swallow exceptions. So, the callsite will never know that the config was not updated. Would urge you to rethink how this is getting handled.

create/update is a simpler model. 

create -> Fails if already exists.
update -> Fails if doesn't exist.

Currently, we have 
* `put` -> Passes always
* `put if not exists` -> Passes always. Swallows exceptions.